### PR TITLE
Add oblt-cli-buildkite-plugin

### PR DESCRIPTION
**Pull request by reakaleek**

### CODE REVIEW

**Comment by reakaleek** [comment]
@v1v, for now, it's only doing the setup. We can add additional commands in follow-ups.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+steps:
+  - label: "E2E Test"
```
Comment (reakaleek) [inline .buildkite/pipeline.yml:2]: Added as a preparation.

When the catalog-info.yaml file is merged into the main branch, we can modify it to run an actual e2e test using the plugin.

File: plugin.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: setup-oblt-cli
+description: Grant pipelines access to Google Cloud resources using Workload Identity Federation
+author: https://github.com/elastic
+public: true
+requirements:
+  - buildkite-agent
```
Comment (v1v) [inline plugin.yml:6]: what's the reason for this requirement?

File: lib/setup.bash [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#   None
+function setup() {
+	local -r version=$1
+	local -r username=$2
+	local -r slack_channel=$3
+	local -r bin_dir=$4
+	local -r asset_id=$(get_asset_id "$version")
+	mkdir -p "${bin_dir}"
+	download_asset "$asset_id" "$bin_dir"
+	export PATH="${bin_dir}:${PATH}"
```
Comment (v1v) [inline lib/setup.bash:28]: Will this export the env variable so it can be used later on in the BK pipeline?

**Review by v1v — APPROVED** [review]
Brilliant, LGTM

File: plugin.yml (hunk 2 above)
Comment (reakaleek) [reply plugin.yml:6]: Oh, I will remove it again, for now.

I was thinking, maybe we want to set the cluster id as meta-data or something.

But this requires a create command or any sort of command, which might be added in the future.

File: lib/setup.bash (hunk 3 above)
Comment (reakaleek) [reply lib/setup.bash:28]: At least, this is what I was hoping.

Still need to create an E2E test where the plugin is actually used.

File: plugin.yml (hunk 2 above)
Comment (reakaleek) [reply plugin.yml]: https://github.com/elastic/oblt-cli-buildkite-plugin/pull/1/commits/44d62d691fe7c91f23c2b3b3381ba60873fd14cf

**Pull request merged**